hack/setup.sh

```diff
@@ -159,6 +181,75 @@ start_registry() {
                 registry:2
 }
 
+start_repository() {
+        log "start repository"
+
+        # Bring up gitea server container
+        docker-compose -f hack/docker-compose.yaml up -d
+
+        # Apply gitea configuration
+        CONTAINER_ID=$(docker ps | cut -d ' ' -f1 | head -2 | tail -1)
+        docker cp hack/golden-app.ini "$CONTAINER_ID:/data/gitea/conf/app.ini"
+        docker restart "$CONTAINER_ID"
+
+        # Create user in server
+        DOCKER_EXEC_SUCCESS=false
+        until $DOCKER_EXEC_SUCCESS
+        do
+          GITEA_TOKEN="$(docker exec -u git "$CONTAINER_ID" gitea admin user create --username "$GIT_WRITER_USERNAME" --password hi --email "$GIT_WRITER_SSH_USER_EMAIL" --admin --access-token --must-change-password=false | head -1 | rev | cut -d ' ' -f1 | rev)" && DOCKER_EXEC_SUCCESS=true || echo "attempting docker exec again" && sleep 5
+        done
+
+        # Generate public/private key
+        ssh-keygen -t rsa -b 4096 -C "$GIT_WRITER_SSH_USER_EMAIL" -f hack/gitea-key -P ""
+        GITEA_KEY_PUB="$(cat hack/gitea-key.pub)"
+
+        # Add the public key to the user on the server
+        CURL_SUCCEED=false
+        until $CURL_SUCCEED
+        do
+          curl -X 'POST' \
+            "http://localhost:3000/api/v1/admin/users/$GIT_WRITER_USERNAME/keys" \
+            -H 'accept: application/json' \
+            -H 'Content-Type: application/json' \
+            -H "Authorization: token $GITEA_TOKEN" \
+            -d "{\"key\": \"$GITEA_KEY_PUB\", \"read_only\": false, \"title\": \"string\"}" \
+            --fail && CURL_SUCCEED=true || echo "retrying curl" && sleep 5
+        done
+
+        # Create a repo on the server
+        curl -X 'POST'   'http://localhost:3000/api/v1/user/repos'   -H 'accept: application/json'   -H 'Content-Type: application/json'   -d '{"name": "'"$GIT_WRITER_REPOSITORY"'", "private": false}' -H "Authorization: token $GITEA_TOKEN"
+
+        GIT_WRITER_SERVER_PUBLIC_TOKEN=${GIT_WRITER_SERVER_PUBLIC_TOKEN:-"$(ssh-keyscan "$PORT_FLAG" "$GIT_WRITER_SERVER")"}
+        echo "$GIT_WRITER_SERVER_PUBLIC_TOKEN" > hack/gitea-server-public-key
+
+        add_ssh_token
+
+        pushd "$(mktemp -d)"
+                touch README.md
+
+                git init
+
+                git config --local user.email "$GIT_WRITER_USERNAME"
+                git config --local user.name "$GIT_WRITER_SSH_USER_EMAIL"
+
+                git add README.md
+                git commit -m "first commit"
+                git remote add origin "$GIT_WRITER_SSH_USER@$GIT_WRITER_SERVER:$GIT_WRITER_SERVER_PORT/$GIT_WRITER_PROJECT/$GIT_WRITER_REPOSITORY.git"
+                git push -u origin master
```

Comment:
```suggestion
                git push -u origin main
```
I had to make this change locally to get this working

Reply:
👍